api/src/transactions.rs

```diff
@@ -770,6 +770,9 @@ impl TransactionsApi {
         if version > ledger_info.version() {
             return Ok(None);
         }
+        if version < ledger_info.oldest_version() {
+            return Ok(None);
+        }
```

Comment:
This should work in most cases.  Remembering that it is possible for the version being used to be pruned prior to processing it (there is an offset though, so it should be fine).